42/README.md

```diff
@@ -70,15 +71,144 @@ The above process ensure the node keeps validating and accepting new blocks whil
+
+Validators vote for the migration:
+
+```
+$ bigchaindb election approve <election-id> --private-key /home/user/.tendermint/config/priv_validator.json
+```
+
+Validators can watch how the election goes:
+
+```
+$ bigchaindb election status <election-id>
+```
+
+Outputs:
+```
+votes_recieved=<Sum_of_votes_recieved>
+votes_allocated=<Sum_of_votes_allocated_in_election>
+network_size=<Total_network_power>
+election_status=<Election_status>
+```
+
+#### 2. Start the new chain
+
+Validators have to install and launch the new version of Tendermint. They need to prepare a new `genesis.json`. The new genesis file has to contain the validator set, the application hash (both at the migration height), and the identifier of the new Tendermint chain (`chain_id`).
+
+`chain_id` is generated and stored by BigchainDB upon conclusion of the migration election. When Tendermint sends this ID as part of the `InitChain` ABCI request, BigchainDB understands that the user has switched to a new Tendermint version so BigchainDB switches to accept new transactions and blocks.
+
+To offer a convenient way to get the data, we propose a CLI command:
+
+```
+$ bigchaindb status
+```
+
+Its output can be extended over time to solve different purposes. For a start, it has to contain the latest known list of validators, the latest known app hash, and the identifier of the Tendermint chain.
+```
+{
+    "chain_id": "...",
+    "validators": [{
+        "pub_key": {
+            "type": "...",
+            "value": "..."
+        },
+        "power": "...",
+        "name": "..."
+    },
+    "app_hash": "..."
+}
+```
+
+Validators upgrading Tendermint are supposed to execute the command above and copy the validators, the app hash, and the chain ID into their `genesis.json`.
+
+Validators joining the network after the migration take the genesis file from existing validators, as usual.
+
+The status command can be always used to see the validator set and the app hash of the latest known block and the identifier of the current Tendermint chain. Note that `chain_id` is only reported after Tendermint sends it as part of the `InitChain` ABCI request.
+
+#### 3. Replay the blockchain
+
+In order to replay the chain up to the migration height, we need to either keep the old Tendermint version running or skip replaying the corresponding part of the chain = archive the chain. We consider the former to be a huge hassle so we describe an archiving scheme further in this BEP.
+
+When a validator replays the chain after a migration, it does not suffice for him to know `genesis.json` - he also needs the archive of the old chain.
+
+Therefore, after the migration election is concluded, each validator has to create a chain archive and keep it together with `genesis.json`. If `genesis.json` is published somewhere, the archive should be published alongside.
+
+To generate the archive, validators can use the `mongodump` command (comes together with `mongod`):
```

Comment:
We now know that the blockchain stored by MongoDB and the blockchain stored by Tendermint are different because Tendermint also stores invalid transactions. Does this matter? Don't the invalid transactions go into calculating some block header hashes on the Tendermint side?

In other words, shouldn't we be archiving Tendermint's LevelDB? Maybe just archive both MongoDB and LevelDB?

(We're considering replacing LevelDB with one MongoDB for everyone. If we do that, this will all become much easier.)

Reply:
No, we do not care about Tendermint storage here by design. After the migration, Tendermint starts from scratch while BigchainDB simply keeps the old data for querying but not for replay.